### DISCUSSION TITLE
OpenDTU_onBattery für SolarFlow und Shelly 3em möglich

### DISCUSSION THREAD

**Discussion by jojo070265** [hoylabs/OpenDTU-OnBattery#524] [Q&A]
Hallo, mich würde interessieren, ob man das ganze Projekt für Zendure Solarflow und die Einbindung von Shelly 3em über HTTP umprogrammieren kann?
Habe von Zendure SolarFlow PV-Hub einen HAM-800 und zwei Zendure Akkus mit zwei 425W Panelen und ein Shelly 3em im Unterverteiler. Bin gerade am Bauen einer OpenDTU und wollte mit dem Shelly 3em direkt eine Phase abgreifen zur OpenDtu schicken und den HM-800 für Battery und Hausgebrauch managen. Ich gehe davon aus, dass alles schon dafür programmiert ist. Aber welche Segmente ich nehmen muss als Anfänger weiß ich noch nicht. Will ohne Homeassistent etc. arbeiten. Man könnte auch vom PV Hub Daten abgreifen. siehe u.a. [solarflow](https://github.com/z-master42/solarflow/tree/main)
/hoymiles.md
Aus jeden Fall Respekt und Dank zu eurer Arbeit.

**Comment by spcqike** [comment]
was genau möchtest du umprogrammieren? Wie funktioniert der SolarFlow PV-Hub?

Ist der SolarFlow "nur" ein Laderegler, der die PV-Solar-Leistung der Module in die Batterie steckt bzw. an seinen Ausgang weiter leitet? Wenn ja, solltest du deine HM-800 "ganz normal" einstecken und laufen lassen können.

den Shelly als PowerMeter konfigurieren, Anleitungen und Beispiele findest du im Wiki. den HM-800 an den SolarFlow. "Fertig".

Ob und wie die Batterie-Überwachungen dann laufen, kann ich dir nicht sagen. ich vermute, gar nicht. die SolarFlow wird die Batterie selbst "schützen" (trennen) und hoffentlich sanft wieder hochfahren, wenn Reserven da sind. Immerhin ist sie quasi eine Art Plug-n-Play Dazwischenstecklösung, um eine Batterie an jedem 0-8-15-Balkonkraftwerk nachzurüsten. Auch ohne externer Steuerung / Nulleinspeisung.

So wie sich dein Link ließt, kann man die SolarFlow auch nicht direkt auslesen. Das geht nur indirekt, per MQTT, welches in der Cloud läuft. Dann gäbe es ein paar Daten. Aber scheinbar kein SoC (State-of-Charge), wenn ich es richtig sehe. Dieser wäre aber sinnvoll. 
Und: OpenDTU kann bisher keine anderen Solardaten verarbeiten, als vom Victron VE.direct. dort gibt es noch keine Möglichkeit für MQTT. 

Wenn du die Hardware bereits hast: Steck es zusammen und teste es einfach. So ist es vom Hersteller ja scheinbar gedacht. Wahrscheinlich mit einem fest gedrosselten Wechselrichter (dauerhaft ~200W Einspeisung? Überschüsse gehen dann in die Batterie)

mit dem Shelly kannst du eine Nulleinspeisung probieren. Sollte das aber zu Schwingen (an/aus der Batterie) oder sonstigem komischen Verhalten führen. drossel den WR einfach auf einen festen Wert. 100W 24/7 sind ja auch 2,4kWh am Tag (sofern das deine Grundlast abbildet)

**Reply by jojo070265** [reply]
Hallo, die openDTU von tbnobody kann keine shelly 3em Daten aufnehmen, die von helgeerbe schon. Diese Aufnahme in die opendtu phasenweise wäre ein Vorteil, um ohne Mqtt Broker; Homeassistent usw. den HM xxx zu limitieren. Das wäre mal aus meiner Sicht ein tolles Addon. Über node-red habe ich schon den shelly über IP ansprechen können und ein Array raus gebracht. Kann es aber nicht rein programmieren.
Viele kaufen sich jetzt ein Micro Solar oder ähnliches, um dann die openDTU einen neuen Wert zu übermitteln. Ich will da nur ein Gerät.
Die Idee den WR auf 100W 24/7 ist werde ich Danke dafür.

**Reply by spcqike** [reply]
Ich glaube, so ganz verstehe ich dich nicht.

Wie gesagt, du kannst den Shelly 3EM direkt in openDTU-onBattery einbinden. du kannst sowohl per MQTT als auch per HTTP(s)+JSON. Damit kennt die oDTU-oB deinen Netzstrombezug.

Dann kannst du auch eine Nulleinspeisung einstellen.

Das Problem: Du kannst deinen Akku nicht so sinnvoll auslesen. Du musst dich quasi mehr oder weniger auf die Spannung verlassen, die am Wechselrichter ankommt (sofern das die Akkuspannung ist, wie gesagt, ich weiß nicht, wie die SolarFlow funktioniert) und musst damit rechnen, dass der SolarFlow irgendwann einfach zu macht, damit die Batterie nicht zu tief entladen wird.

> Über node-red habe ich schon den shelly über IP ansprechen können und ein Array raus gebracht. Kann es aber nicht rein programmieren.

was genau möchtest du denn als Array wo haben? und warum?

> Diese Aufnahme in die opendtu phasenweise wäre ein Vorteil, um ohne Mqtt Broker; Homeassistent usw. den HM xxx zu limitieren. Das wäre mal aus meiner Sicht ein tolles Addon

wie gesagt, du kannst die Leistung phasenweise auswerten. mit MQTT oder mit HTTP(S)+JSON Abfrage.

**Reply by 16RP72** [reply]
Ich glaube er sucht die Kopplung zur Zenture Einheit im CT Modus, würde mich auch interessieren, da ich eine Regelung über die Cloud nicht gut finde.
